### ISSUE TITLE
"Move Destabilized"

### ISSUE THREAD

**Issue by Nirvanana** [cboulay/PSMoveService#130]
When I was trying to calibrate magnetometer of the second controller it kept showing "Move Destabilized" though my controller was placed on the table pointing upward. 
The first controller worked. Probably the second controller is broken? 
But I can see its orientation in the test, I just cannot calibrate it, neither when I want to calibrate the pose.

**Comment by HipsterSloth** [comment]
In order to make sure that the controller is stable on the table so that I'm getting a reliable reading I currently make sure the accelerometer reading is pointing straight down (+1g). It's possible that if the accelerometer is skewed ever so slightly in your PSMove that when you have it level on your table the accelerometer reading isn't close enough to down (I have some small 10 degree tolerance). You might be able to hold the controller firm and tilt it ever so slightly in one way or another to get the gravity direction to line up.

That said, I think the real fix for this it for me to record what "down" is first rather than assume what down should be. I'm updating the bug title to reflect this.

**Comment by Nirvanana** [comment]
Thanks a lot!
I tried many positions but it still didn't work.
I will try more times and I will update here if it works.

**Comment by cboulay** [comment]
@Nirvanana Any update on this? Can I close the issue?

**Comment by rodrigost23** [comment]
@cboulay Seeing they did not answer, I'd like to say that I have the same problem, so I think it's still a valid issue.

**Comment by HipsterSloth** [comment]
Yeah I haven't gotten around to implementing this yet unfortunately. It's not hard, I'v just been getting pulled in many directions. Maybe @gb2111 might be interesting working on this. Greg has implemented a couple of other ConfigTool improvements so far.

**Comment by gb2111** [comment]
so that would be basically 'thrust me, its stable' button?

**Comment by gb2111** [comment]
Greetings, 

Have  you heard the latest news? Not yet?  You have to read it  then, because  it's just woooow! Here is the info <http://dream.nevadawholesalekoi.com/yd/04>

Regards, knurgb

**Comment by rodrigost23** [comment]
Does the commit 8e1c732d368f44a105f24bf60506de3ed36508c9 have anything to do with this issue?

**Comment by HipsterSloth** [comment]
@rodrigost23 No that check-in has to do with providing a way to reset the orientation of the controller if it's visual orientation had drifted from it's real world orientation. I should add the "Trust me, it's stable" button for this case too. I want to put out another official release soon with a bunch of magnetometer related fixes in it.

**Comment by rodrigost23** [comment]
I don't know the priority of this or other issues above it, but this still hasn't been resolved. And I can't find any workarounds to calibrate the Move, as I have tried holding it still in different directions.

**Comment by HipsterSloth** [comment]
Sorry this slipped off my radar for a while, but I actually ran into this issue again myself yesterday at work. We wanted to try using PSMoveService at my day job again and this is blocking our calibration. I have a few  issues related to that I wanted to fix this weekend as part of finalizing DualShock4 support.

**Comment by HipsterSloth** [comment]
@rodrigost23 I didn't have time this weekend to implement the full solution (where we calibrate the down direction in the accelerometer calibration), but I was able to add a "Trust me, it's stable" button to the magnetometer calibration screen. Would you mind giving this custom build a try? If it works for you I'll upload a new release.

[PSMoveService-v0.9alpha6_0_11_custom.zip](https://github.com/cboulay/PSMoveService/files/722935/PSMoveService-v0.9alpha6_0_11_custom.zip)

**Comment by rodrigost23** [comment]
@HipsterSloth I just tried it, and the button worked for calibrating the magnetometer -- still not working for gyroscope calibration ;)

**Comment by HipsterSloth** [comment]
Technically you don't need to calibrate the gyroscope, but I figured it couldn't hurt to add the same button in the gyro calibration tool as well:

[PSMoveService-v0.9alpha6_0_11b_custom.zip](https://github.com/cboulay/PSMoveService/files/725789/PSMoveService-v0.9alpha6_0_11b_custom.zip)
